packages/back-end/src/jobs/sdkWebhooks.ts

```diff
@@ -161,29 +161,42 @@ async function runWebhookFetch({
   const signingKey = webhook.signingKey;
   const headers = webhook.headers || "";
   const method = webhook.httpMethod || "POST";
-  const sendPayload = webhook.sendPayload;
+  // todo: better inference based on sendPayload
+  const payloadFormat = webhook.payloadFormat || "standard";
   const organizationId = webhook.organization;
   const requestTimeout = 30000;
   const maxContentSize = 1000;
 
+  const sendPayload =
+    method !== "GET" && ["standard", "sdkPayload"].includes(payloadFormat);
+
   const date = new Date();
   const signature = createHmac("sha256", signingKey)
     .update(sendPayload ? payload : "")
     .digest("hex");
   const secret = `whsec_${signature}`;
   const webhookID = `msg_${md5(key + date.getTime()).substr(0, 16)}`;
-  const data = sendPayload ? { payload } : {};
 
   const timestamp = Math.floor(date.getTime() / 1000);
 
-  const body =
-    method === "GET"
-      ? undefined
-      : JSON.stringify({
-          type: "payload.changed",
-          timestamp: date.toISOString(),
-          data,
-        });
+  let body: string | undefined = undefined;
+
+  if (method !== "GET") {
+    if (payloadFormat === "standard") {
+      body = JSON.stringify({
+        type: "payload.changed",
+        timestamp: date.toISOString(),
+        data: { payload },
+      });
+    } else if (payloadFormat === "standard-no-payload") {
+      body = JSON.stringify({
+        type: "payload.changed",
+        timestamp: date.toISOString(),
+      });
+    } else if (payloadFormat === "sdkPayload") {
+      body = payload;
+    }
```

Comment:
You could use the extensible switch form here to make sure the code is updated whenever a new format is added:
```typescript
let invalidFormat: never;
switch (payloadFormat) {
  case "standard"
      body = JSON.stringify({
        type: "payload.changed",
        timestamp: date.toISOString(),
        data: { payload },
      });
      break;
      ...
    
  // Should have exhausted all possibilities here:
  default:
    invalidFormat = format;
     // Should never happen:
   throw new Error(`Invalid format: ${invalidFormat}`);
}

Reply:
I'm not comfortable throwing an error here. I'd much rather default to something sensible. But I can migrate this to a switch statement

Reply:
The error is never sent. You can put a sensitive returned value if you prefer but if the code ever reach it, something is wrong in the underlying assumptions.

More details:

You have two situations:
* The `case` above have exhausted all possible types for `format`, there is no possible type for `format` so it is in fact of type `never` in this branch and it works but `default` is never executed
* The `case` above are missing one variant for `format` an the type-checker raises an error.

Here's a typescript playground with a smaller version for illustration: https://www.typescriptlang.org/play/?#code/C4TwDgpgBAYg9nKBeKAiARgGwIaqgHzQHMA7AS1QChKB6GqAFQAsyBnKOE6NqAWzdZkSRKNigBjbKwgAuSuM6tgUIQt5hMEYBAAKAJzjiIrVgEFMmeIhQAKAB5QZsBAEpkAPigBvSlCiblIQA3bEwyABMANVCAV1koLiCIPQBuaj9WAHcyYHEmKHs3Hz8-SWk0LFw5EpK9LRi9EjQcqEypUShK1DSa8IgAM2wYzGBqmpUSELCo2OgUOx7x4CYDTISINYBRPQM9GwADAElJ0IioKbinABIvYNOZzDiAX32XRafKD9p6Zh5ObnYRAQ4TkChISigYAMRhM5ksCGQBQcTisbiQnmK-i0EymEWij3iiWSaV8UCyOTySKKpNKUmgGBwqDGNTqwAaTVQLTa7DEXUWNTK9NIFGZtXqjWaym5HT5NKgfUGw1Gcr8d2m+LiiIWKqgy1W6y2OzgeyOJ2m51m11uZrxsxeb1JHyeQA

The idea of this pattern is to make sure that the typechecker can detect unhandled cases in the future. Type for instance to add a case to the type `Foo` in the above playground example..

Reply:
We shouldn't throw an error here. Format can be something outside of the allowed types because one entrypoint is raw parsing of an environment variable (for global webhooks). They don't go through the standard zod validators and migration.

I added a missing case guard though, similar to your suggestion.